### PR TITLE
[path_provider] Update the Linux example template

### PR DESCRIPTION
**Pull request by robert-ancell**
Minor change, no issue opened.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/path_provider/path_provider_linux/example/linux/flutter/CMakeLists.txt [hunk 1]
```diff
@@ -80,7 +80,8 @@ add_custom_command(
   COMMAND ${CMAKE_COMMAND} -E env
     ${FLUTTER_TOOL_ENVIRONMENT}
     "${FLUTTER_ROOT}/packages/flutter_tools/bin/tool_backend.sh"
-      linux-x64 ${CMAKE_BUILD_TYPE}
+      ${FLUTTER_TARGET_PLATFORM} ${CMAKE_BUILD_TYPE}
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider_linux/example/linux/flutter/CMakeLists.txt:83]: This isn't valid on the stable channel.

Comment (stuartmorgan) [reply packages/path_provider/path_provider_linux/example/linux/flutter/CMakeLists.txt:83]: @robert-ancell Ping; is there a specific issue this was trying to solve? This should auto-migrate for people when building on master.

**Comment by stuartmorgan** [comment]
Closing for now; if there's a specific need to land something like this before the changes reach stable we'll need to figure out how to address that, but it should be fine to just wait AFAIK.

File: packages/path_provider/path_provider_linux/example/linux/flutter/CMakeLists.txt (hunk 1 above)
Comment (robert-ancell) [reply packages/path_provider/path_provider_linux/example/linux/flutter/CMakeLists.txt:83]: This PR was just to test if the CI build failures were due to https://github.com/flutter/plugins/pull/3077. I thought this change was trivial but apparently not trivial enough :)

**Pull request closed without merging**